src/v/config/configuration.cc

```diff
@@ -708,7 +722,10 @@ configuration::configuration()
       {.needs_restart = needs_restart::no,
        .visibility = visibility::user,
        .aliases = {"delete_retention_ms"}},
-      7 * 24h)
+      7 * 24h,
+      property<std::optional<std::chrono::milliseconds>>::noop_validator,
+      std::nullopt,
+      retention_ms_constraint_methods)
```

Comment:
> Can we consider making all of the constructor take a plain struct so we can name all the fields and what they mean?

I took a stab at this in https://github.com/NyaliaLui/redpanda/commit/0e63c60df450171bb06525624406d862ddc9632f and made a ticket for this at https://github.com/redpanda-data/redpanda/issues/14621. This is a QoL improvment but it could be really helpful.

Reply:
Reordered constraint_methods that minimizes the blast radius but also retains argument context.

Reply:
Thanks, honestly this would have been good opportunity to add a new constructor that took an args struct so we could use named args for almost everything here.

Reply:
Wrapping config params in a struct is a great QoL improvement but it is out-of-scope of this project.